Upstream Pencil Hover

megangardner opened the pull request:
#### 9856bc8b756610ac4e3a4a2c480e6332f50e0cd7
<pre>
Upstream Pencil Hover
<a href="https://bugs.webkit.org/show_bug.cgi?id=247085">https://bugs.webkit.org/show_bug.cgi?id=247085</a>
rdar://88513609

Reviewed by Tim Horton.

* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/html/HTMLTextFormControlElement.cpp:
(WebCore::pointerTypeFromHitTestRequest):
(WebCore::HTMLTextFormControlElement::placeholderShouldBeVisible const):
(WebCore::showPlaceholderForPointer): Deleted.
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::handleMouseMoveEvent):
(addPointerTypeToHitTestRequest): Deleted.
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(shouldEnableAlternativeMouseGestureRecognizers):
* Source/WebKit/UIProcess/ios/WKMouseGestureRecognizer.mm:
(pointerTypeForUITouchType):

Canonical link: <a href="https://commits.webkit.org/256074@main">https://commits.webkit.org/256074@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5d6542ffd0fdd6089f8644502012b8a6e424e0f5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94515 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3693 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/27421 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104169 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/164441 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/98513 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3745 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31889 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86843 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/100140 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100185 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/2693 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80906 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29720 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84618 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/84191 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/72614 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebView:/webkit/WebKitWebView/is-web-process-responsive (failure)") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/85719 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38290 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/18013 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/80885 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36154 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/19286 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/27800 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4196 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40053 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/41925 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/83553 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42006 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38516 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/18877 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -513,6 +513,10 @@
 #endif
 
+#if !defined(ENABLE_PENCIL_HOVER) && PLATFORM(IOS_FAMILY)
+#define ENABLE_PENCIL_HOVER 0
```
Seems like this should not be 0

hortont424 commented on Source/WebCore/html/HTMLTextFormControlElement.cpp:
```diff
@@ -117,6 +100,15 @@ Node::InsertedIntoAncestorResult HTMLTextFormControlElement::insertedIntoAncesto
 
+static String pointerTypeFromHitTestRequest(HitTestRequest request)
+{
+    if (request.penEvent())
```
Should some part of this be #if'd?

hortont424 commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -10118,6 +10100,22 @@ - (void)mouseGestureRecognizerChanged:(WKMouseGestureRecognizer *)gestureRecogni
+static bool shouldEnableAlternativeMouseGestureRecognizers(WKMouseGestureRecognizer *mouseRecognizer, WKMouseGestureRecognizer *alternativeMouseRecognizer)
+{
+#if ENABLE(PENCIL_HOVER)
+    [alternativeMouseRecognizer setAllowedTouchTypes:@[@(UITouchTypePencil)]];
```
Please rename "alternative"

hortont424 replied:
Also I think you want some spaces around your @[ ]: `@[ @(UITouchTypePencil) ]`.

megangardner committed:
Upstream Pencil Hover
https://bugs.webkit.org/show_bug.cgi?id=247085
rdar://88513609

Reviewed by Tim Horton.

* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/html/HTMLTextFormControlElement.cpp:
(WebCore::pointerTypeFromHitTestRequest):
(WebCore::HTMLTextFormControlElement::placeholderShouldBeVisible const):
(WebCore::showPlaceholderForPointer): Deleted.
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::handleMouseMoveEvent):
(addPointerTypeToHitTestRequest): Deleted.
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(shouldEnableAlternativeMouseGestureRecognizers):
* Source/WebKit/UIProcess/ios/WKMouseGestureRecognizer.mm:
(pointerTypeForUITouchType):

Canonical link: https://commits.webkit.org/256074@main
